src/main/java/amazin/service/SecurityServiceImpl.java

```diff
@@ -22,9 +23,11 @@
 
     @Override
     public String findLoggedInEmail() {
-        Object userDetails = SecurityContextHolder.getContext().getAuthentication().getDetails();
-        if (userDetails instanceof UserDetails) {
-            return ((UserDetails)userDetails).getUsername();
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+        if (authentication.getPrincipal() instanceof UserDetails) {
```

Comment:
Can you please explain what was wrong with the initial implementation?

Reply:
The initial implementation, even when the user was logged in, for some reason was always returning null. This modification properly returns back the email if the user is logged in or null as the email if no user is logged in.